src/ir_Sharp.cpp

```diff
@@ -339,7 +340,20 @@ void IRSharpAc::on(void) { setPower(true); }
 void IRSharpAc::off(void) { setPower(false); }
 
 void IRSharpAc::setPower(const bool on) {
-  setBit(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, on);
+  if (on)
+    setBits(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, 2, kSharpAcOnFromOff);
+  else
+    setBits(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, 2, kSharpAcOffFromOn);
+}
+
+void IRSharpAc::setPower(const bool on, const bool prev) {
+  setBit(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, !on);
+  if (on && prev)
+    setBits(&remote[kSharpAcBytePower], kSharpAcBitPowerOffset, 2, kSharpAcOnFromOn);
```

Comment:
`src/ir_Sharp.cpp:352:  Lines should be <= 80 characters long  [whitespace/line_length] [2]`